mycli/packages/completion_engine.py

```diff
@@ -28,32 +28,27 @@ def suggest_type(full_text, text_before_cursor):
 
     identifier = None
 
-    # This is a temporary hack; the exception handling
-    # here should be removed once sqlparse has been fixed
-    try:
-        # If we've partially typed a word then word_before_cursor won't be an empty
-        # string. In that case we want to remove the partially typed string before
-        # sending it to the sqlparser. Otherwise the last token will always be the
-        # partially typed string which renders the smart completion useless because
-        # it will always return the list of keywords as completion.
-        if word_before_cursor:
-            if word_before_cursor[-1] == '(' or word_before_cursor[0] == '\\':
-                parsed = sqlparse.parse(text_before_cursor)
-            else:
-                parsed = sqlparse.parse(
-                        text_before_cursor[:-len(word_before_cursor)])
+    # If we've partially typed a word then word_before_cursor won't be an empty
+    # string. In that case we want to remove the partially typed string before
+    # sending it to the sqlparser. Otherwise the last token will always be the
+    # partially typed string which renders the smart completion useless because
+    # it will always return the list of keywords as completion.
+    if word_before_cursor:
+        if word_before_cursor[-1] == '(' or word_before_cursor[0] == '\\':
```

Comment:
I know this was already like this, but we should change this to use `str.startswith()` and `str.endswith()`.